plutus-core/plutus-ir/src/PlutusIR/Transform/Inline.hs

```diff
@@ -152,48 +158,60 @@ type InliningConstraints tyname name uni fun =
     , PLC.ToBuiltinMeaning uni fun
     )
 
-
+-- | Useful info for inlining.
 data InlineInfo name fun a = InlineInfo
     { _iiStrictnessMap :: Deps.StrictnessMap
+    -- ^ Is it strict? Only needed for PIR, not UPLC
     , _iiUsages        :: Usages.Usages
+    -- ^ how many times is it used?
     , _iiHints         :: InlineHints name a
+    -- ^ have we explicitly been told to inline.
     , _iiBuiltinVer    :: PLC.BuiltinVersion fun
+    -- ^ the builtin version.
     }
 makeLenses ''InlineInfo
 
--- Using a concrete monad makes a very large difference to the performance of this module (determined from profiling)
-type InlineM tyname name uni fun a = ReaderT (InlineInfo name fun a) (StateT (Subst tyname name uni fun a) Quote)
+-- | The monad the inliner runs in.
+-- Using a concrete monad makes a very large difference to the performance of this module
```

Comment:
It's a bit subtle, but what I've tended to do is to leave some comments as non-haddock comments, if they're only interesting for people looking at the _definition_ of the name. i.e. do you want to see this when you hover over the name? Probably not. It's useful if you're looking at the definition, though! So I'd be tempted to do:
```
-- Using a concrete monad ...
-- | The monad the inliner runs in.
```